lib/screens/screens_by_alert/self_refresh_runner.ex

```diff
@@ -0,0 +1,102 @@
+defmodule Screens.ScreensByAlert.SelfRefreshRunner do
+  @moduledoc """
+  A "job-runner" GenServer that routinely checks for out-of-date
+  screen data, and simulates data requests for any that it finds.
+  """
+
+  alias Screens.ScreensByAlert
+
+  # (Not a real module--just a name assigned to the Task.Supervisor process that supervises each simulated data request run)
+  alias Screens.ScreensByAlert.SelfRefreshRunner.TaskSupervisor
+
+  require Logger
+  use GenServer
+
+  @type state :: %__MODULE__{}
+
+  # The server is not stateful.
+  defstruct []
+
+  ### Client
+
+  def start_link(opts) do
+    GenServer.start_link(__MODULE__, :ok, opts)
+  end
+
+  ### Server
+
+  # Maximum number of screen updates that can happen per run.
+  # Assuming a worst case of .5 sec per screen update, a max
+  # of 20 has the job take 10 seconds max per run, giving
+  # plenty of space to avoid overlapping runs.
+  @max_screen_updates_per_run 20
+
+  @screens_ttl_seconds Application.compile_env(:screens, :screens_by_alert)[:screens_ttl_seconds]
+
+  # The job runs at the same rate as screen data expiration from the cache (though these "windows" will be offset from one another)
+  @job_run_interval_ms @screens_ttl_seconds * 1_000
+
+  @screen_data_fn Application.compile_env(:screens, :screens_by_alert)[:screen_data_fn]
+
+  @impl true
+  def init(:ok) do
+    {:ok, %__MODULE__{}, {:continue, :schedule_first_run}}
+  end
+
+  @impl true
+  def handle_continue(:schedule_first_run, state) do
+    schedule_run()
+
+    {:noreply, state}
+  end
+
+  @impl true
+  def handle_info(:run, state) do
+    schedule_run()
+
+    now = System.system_time(:second)
+
+    {screen_ids_to_refresh, overflow} =
+      Screens.Config.State.v2_screens_visible_to_screenplay()
+      # get a mapping from each ID to its last updated time
+      |> ScreensByAlert.get_screens_last_updated()
+      # filter to outdated IDs
+      |> Enum.filter(fn {_screen_id, timestamp} -> now - timestamp > @screens_ttl_seconds end)
```

Comment:
I think you could make `Enum.filter` and `Enum.map` below into `Stream.filter` and `Stream.map`? Would save us two iterations through the map, but probably barely makes a difference performance wise 🤷

Reply:
When I've informally tested it out before, `Stream` has only conferred performance gains when we might stop short of using all of the values that the stream spits out, or when the thing we're iterating over is huge. In this case, it's a finite and not-too-big enumerable, and we want to use all of the elements produced, so I'm sticking with `Enum` functions.

Reply:
Sounds good to me!